Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -274,67 +300,175 @@ def TestConfigUpdate(Contents):
             retval = -1
     return retval
 
-def SetConfigUpdate(Contents):
-    destFileFullPath = CONFIG_PATH.__add__(DEST_FILE_NAME)
+def TestMultiHoming(ConfigID, Ensure):
+    # Check if state directory exists if Ensure is Present
+    if Ensure == 'Present' and not os.path.exists(PATH_NPM_STATE):
+        LG().Log(LogType.Info, 'Workspace has NPM enabled but state directory absent, set will create')
+        return -1
+
+    # Get workspace lists
+    wsListFromState = GetWorkspaceListFromStateFile()
+    wsListFromDirs = GetWorkspaceListFromDirectories()
+
+    # Check if state file has to be altered
+    for x in wsListFromState:
+        if x not in wsListFromDirs:
+            LG().Log(LogType.Info, 'Workspaces either onboarded/removed, set to rectify state file')
```

Comment:
If a workspace is in an outdated state file but not in the directory structure, then it must have been removed, not onboarded.

The opposite check (for every x in wsListFromDirs, if it is not in wsListFromState) indicates that workspace has been recently onboarded.

You should make both checks to log exactly what happened, if you really want to ensure that the lists are the same.